src/time/format/formatter.cr

```diff
@@ -148,9 +148,22 @@ struct Time::Format
     end
 
     def second_fraction?(fraction_digits = nil)
-      unless time.nanosecond == 0 || fraction_digits == 0
-        char '.'
-        second_fraction
+      if fraction_digits
+        if fraction_digits > 0
```

Comment:
Shouldn't it raise if fraction_digits < 0?

Reply:
Done